module/zfs/zil.c

```diff
@@ -1870,6 +1871,25 @@ zil_lwb_write_close(zilog_t *zilog, lwb_t *lwb)
 
 	dmu_tx_commit(tx);
 
+	/*
+	 * Acquire config lock for the lwb to issue it later.  If we have a
+	 * queue of already closed parent lwbs already holding config lock,
+	 * we can not block here now, or it will be a deadlock.
```

Comment:
Could you expand/clarify this comment a bit? Perhaps something like:
```
* We need to acquire the config lock for the lwb to issue it later. However, if
* we already have a queue of closed parent lwbs already holding the config
* lock (but not yet issued), we can't block here waiting on the lock or we will
* deadlock. In that case we must first issue to parent IOs before waiting on 
* the lock.
```
As an aside: why don't we always just issue the IOs in `ilwbs` here? Is it less efficient?

Reply:
> As an aside: why don't we always just issue the IOs in ilwbs here? Is it less efficient?

see commit f63811f07213361d49878648bd597af88d06859c that moved it out from under the `zl_issuer_lock`